### PR TITLE
Cleanup Gradle dependencies

### PR DESCRIPTION
**Pull request by Wyvest**
## Description
Organizes all our dependencies to either settings.gradle.kts (for PGT) and libs.version.toml (for everything else)

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

**Review by pauliesnug — APPROVED** [review]
lgtm (you stole this from me)

**Review by nxtdaydelivery — APPROVED** [review]
so pretty

File: build.gradle.kts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
 plugins {
-    kotlin("jvm") version "1.6.21"
-    id("org.jetbrains.kotlinx.binary-compatibility-validator") version "0.12.1"
-    id("cc.polyfrost.defaults.repo")
-    id("cc.polyfrost.defaults.java")
-    id("net.kyori.blossom") version "1.3.0"
+    alias(libs.plugins.kotlin)
+    alias(libs.plugins.kotlinAbi)
+    alias(pgtLibs.plugins.pgtDefaultJava)
```
Comment (xtrm-en) [inline build.gradle.kts:12]: maybe `pgtDefaultJava` -> `defaultJava` since it's already in pgtLibs

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# PGT-related stuff is in settings.gradle.kts
+[versions]
+kotlin = "1.6.21"
```
Comment (xtrm-en) [inline gradle/libs.versions.toml:3]: crying emoji (im sure theres a way to make it work)

File: versions/build.gradle.kts [hunk 3]
```diff
@@ -511,6 +490,10 @@ fun DependencyHandlerScope.include(
     }
 }
 
+fun DependencyHandlerScope.include(dependency: Provider<MinimalExternalModuleDependency>, pom: Boolean = true, mod: Boolean = false, relocate: Boolean = false, transitive: Boolean = true) {
+    include(dependency.get().run { "$group:$name:$version" }, pom, mod, relocate, transitive)
+}
```
Comment (xtrm-en) [inline versions/build.gradle.kts:495]: pain and suffering

**Review by xtrm-en — APPROVED** [review]
lgtm

**Pull request merged**